es-core/src/GuiComponent.cpp

```diff
@@ -105,6 +105,26 @@ void GuiComponent::setSize(float w, float h)
     onSizeChanged();
 }
 
+float GuiComponent::getZIndex() const
+{
+	return zIndex;
+}
+
+void GuiComponent::setZIndex(float z)
+{
+	zIndex = z;
+}
+
+float GuiComponent::getDefaultZIndex() const
+{
+	return zIndex;
+}
+
+void GuiComponent::setDefaultZIndex(float z)
+{
+	zIndex = z;
+}
+
```

Comment:
But now these two pairs of functions return/do exactly the same thing, right? So what is the difference?At some other location, the GuiComponent is instantiated with a default value, which will remain the same if we do not explicitly set it.